packages/compass-smoke-tests/src/cli.ts

```diff
@@ -67,51 +80,116 @@ const argv = yargs(hideBin(process.argv))
-    type: 'string',
-    choices: SUPPORTED_PACKAGES,
-    demandOption: true,
-    description: 'Which package to test',
-  })
-  .option('forceDownload', {
-    type: 'boolean',
-    description: 'Force download all assets before starting',
-  })
-  .option('localPackage', {
-    type: 'boolean',
-    description: 'Use the local package instead of downloading',
-  })
-  .option('skipCleanup', {
-    type: 'boolean',
-    description: 'Do not delete the sandboxes after a run',
-    default: false,
-  })
-  .option('skipUninstall', {
-    type: 'boolean',
-    description: 'Do not uninstall after a run',
-    default: false,
-  })
-  .option('tests', {
-    type: 'array',
-    string: true,
-    choices: SUPPORTED_TESTS,
-    description: 'Which tests to run',
-  })
-  .default('tests', SUPPORTED_TESTS.slice());
-
-async function run() {
-  const context: SmokeTestsContext = argv.parseSync();
+  .command(
+    '$0',
+    'Run smoke tests',
+    (argv) =>
+      argv
+        .option('platform', {
+          choices: SUPPORTED_PLATFORMS,
+          demandOption: true,
+          default: getDefaultPlatform(),
+        })
+        .option('arch', {
+          choices: SUPPORTED_ARCHS,
+          demandOption: true,
+          default: getDefaultArch(),
+        })
+        .option('package', {
+          type: 'string',
+          choices: SUPPORTED_PACKAGES,
+          demandOption: true,
+          description: 'Which package to test',
+        })
+        .option('forceDownload', {
+          type: 'boolean',
+          description: 'Force download all assets before starting',
+        })
+        .option('localPackage', {
+          type: 'boolean',
+          description: 'Use the local package instead of downloading',
+        })
+        .option('skipCleanup', {
+          type: 'boolean',
+          description: 'Do not delete the sandboxes after a run',
+          default: false,
+        })
+        .option('skipUninstall', {
+          type: 'boolean',
+          description: 'Do not uninstall after a run',
+          default: false,
+        })
+        .option('tests', {
+          type: 'array',
+          string: true,
+          choices: SUPPORTED_TESTS,
+          description: 'Which tests to run',
+        })
+        .default('tests', SUPPORTED_TESTS.slice()),
+    async (args) => {
+      await run(args);
+    }
+  )
+  .command(
+    'dispatch',
+    'Dispatch smoke tests on GitHub and watch to completion',
+    (argv) =>
+      argv
+        .option('github-pr-number', {
+          type: 'number',
+          description: 'GitHub PR number used to determine ref',
+        })
+        .option('ref', {
+          type: 'string',
+          description: 'Git reference to dispatch the workflow from',
+          default: getDefaultRef(),
+        })
+        .option('version', {
+          type: 'string',
+          description: 'Compass version to run tests for',
+          default: process.env.DEV_VERSION_IDENTIFIER,
```

Comment:
You're going to have to get version from hadron-build info, I think. So it will work for beta and stable releases and not just PRs or dev builds.

Reply:
Currently when smoke testing it forms part of the filename that is in type PackageDetails. So we don't have an explicit version. You seem to only be using it for the task name so you can probably come up with something that just means beta or stable in that case? There might even be an env or evergreen var already 🤔